PWGDQ/Core/VarManager.h

```diff
@@ -1029,6 +1041,144 @@ void VarManager::FillPairVertexing(C const& collision, T const& t1, T const& t2,
+    o2::track::TrackParCov pars1{lepton1.x(), lepton1.alpha(), lepton1pars, lepton1covs};
+    std::array<float, 5> lepton2pars = {lepton2.y(), lepton2.z(), lepton2.snp(), lepton2.tgl(), lepton2.signed1Pt()};
+    std::array<float, 15> lepton2covs = {lepton2.cYY(), lepton2.cZY(), lepton2.cZZ(), lepton2.cSnpY(), lepton2.cSnpZ(),
+                                         lepton2.cSnpSnp(), lepton2.cTglY(), lepton2.cTglZ(), lepton2.cTglSnp(), lepton2.cTglTgl(),
+                                         lepton2.c1PtY(), lepton2.c1PtZ(), lepton2.c1PtSnp(), lepton2.c1PtTgl(), lepton2.c1Pt21Pt2()};
+    o2::track::TrackParCov pars2{lepton2.x(), lepton2.alpha(), lepton2pars, lepton2covs};
+    std::array<float, 5> lepton3pars = {track.y(), track.z(), track.snp(), track.tgl(), track.signed1Pt()};
+    std::array<float, 15> lepton3covs = {track.cYY(), track.cZY(), track.cZZ(), track.cSnpY(), track.cSnpZ(),
+                                         track.cSnpSnp(), track.cTglY(), track.cTglZ(), track.cTglSnp(), track.cTglTgl(),
+                                         track.c1PtY(), track.c1PtZ(), track.c1PtSnp(), track.c1PtTgl(), track.c1Pt21Pt2()};
+    o2::track::TrackParCov pars3{track.x(), track.alpha(), lepton3pars, lepton3covs};
+    procCode = VarManager::fgFitterThreeProngBarrel.process(pars1, pars2, pars3);
+    procCodeJpsi = VarManager::fgFitterTwoProngBarrel.process(pars1, pars2);
+  } else {
+    return;
+  }
+
+  ROOT::Math::PtEtaPhiMVector v1(lepton1.pt(), lepton1.eta(), lepton1.phi(), mtrack);
+  ROOT::Math::PtEtaPhiMVector v2(lepton2.pt(), lepton2.eta(), lepton2.phi(), mtrack);
+  ROOT::Math::PtEtaPhiMVector v3(track.pt(), track.eta(), track.phi(), mtrack);
+  ROOT::Math::PtEtaPhiMVector v12 = v1 + v2;
+  ROOT::Math::PtEtaPhiMVector vdilepton(v12.pt(), v12.eta(), v12.phi(), v12.M());
+  ROOT::Math::PtEtaPhiMVector v123 = vdilepton + v3;
+  values[VarManager::kPairMass] = v123.M();
+  values[VarManager::kPairPt] = v123.Pt();
+  values[VarManager::kPairEta] = v123.Eta();
+
+  values[VarManager::kPairMassDau] = v12.M();
+  values[VarManager::kPairPtDau] = v12.Pt();
+  values[VarManager::kPt] = track.pt();
+
+  values[VarManager::kVertexingProcCode] = procCode;
+  if (procCode == 0 || procCodeJpsi == 0) {
+    // TODO: set the other variables to appropriate values and return
+    values[VarManager::kVertexingChi2PCA] = -999.;
+    values[VarManager::kVertexingLxy] = -999.;
+    values[VarManager::kVertexingLxyz] = -999.;
+    values[VarManager::kVertexingLz] = -999.;
+    values[VarManager::kVertexingLxyErr] = -999.;
+    values[VarManager::kVertexingLxyzErr] = -999.;
+    values[VarManager::kVertexingLzErr] = -999.;
+
+    values[VarManager::kVertexingTauxy] = -999.;
+    values[VarManager::kVertexingTauz] = -999.;
+    values[VarManager::kVertexingTauxyErr] = -999.;
+    values[VarManager::kVertexingTauzErr] = -999.;
+    return;
+  }
+
+  Vec3D secondaryVertex;
```

Comment:
The secondary vertex does not seem to be computed anywhere, this is just zero i guess. So the rest of the quantities are not valid at the moment, right? You would probably need to also get the secondary vertex from the two leptons, because we could compute a few quantities from that and used them for selections.
You need also to add the uncertainties on the L values, similar to what is done in the two prong approach.

Reply:
Fixed with the new commit